service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DocumentReferenceToNarrativeStatementMapper.java

```diff
@@ -0,0 +1,154 @@
+        final Optional<CodeableConcept> codeableConcept = Optional.ofNullable(documentReference.getContext())
+            .map(DocumentReference.DocumentReferenceContextComponent::getPracticeSetting);
+
+        codeableConcept
+            .filter(CodeableConcept::hasText)
+            .map(CodeableConcept::getText)
+            .ifPresentOrElse(text -> commentBuilder.append("Setting: ").append(text).append(StringUtils.SPACE),
+                () -> codeableConcept
+                    .filter(CodeableConcept::hasCoding)
+                    .map(CodeableConcept::getCodingFirstRep)
+                    .filter(Coding::hasDisplay)
+                    .map(Coding::getDisplay)
+                    .ifPresent(display -> commentBuilder.append("Setting: ").append(display).append(StringUtils.SPACE)));
+    }
+
+    private void mapDescription(final DocumentReference documentReference, final StringBuilder commentBuilder) {
+        Optional.ofNullable(documentReference.getDescription())
+            .ifPresent(desc -> commentBuilder.append("Description: ").append(desc).append(StringUtils.SPACE));
+    }
+
+    private void mapCustodianOrg(final DocumentReference documentReference, final StringBuilder commentBuilder) {
+        Optional.ofNullable(documentReference.getCustodian())
+            .map(BaseReference::getReferenceElement)
+            .filter(IIdType::hasResourceType)
+            .filter(idType -> idType.getResourceType().equals(ResourceType.Organization.name()))
+            .ifPresent(idType -> messageContext.getInputBundleHolder()
+                .getResource(idType)
+                .map(Organization.class::cast)
+                .map(Organization::getName)
+                .map(name -> commentBuilder.append("Custodian Org: ").append(name).append(StringUtils.SPACE)));
+    }
+
+    private void mapAuthorOrg(final DocumentReference documentReference, final StringBuilder commentBuilder) {
+        documentReference.getAuthor().stream()
+            .map(BaseReference::getReferenceElement)
+            .filter(IIdType::hasResourceType)
+            .filter(idType -> idType.getResourceType().equals(ResourceType.Organization.name()))
+            .findFirst()
+            .ifPresent(idType -> messageContext.getInputBundleHolder()
+                .getResource(idType)
+                .map(Organization.class::cast)
+                .map(Organization::getName)
+                .map(name -> commentBuilder.append("Author Org: ").append(name).append(StringUtils.SPACE)));
+    }
+
+    private void mapType(final DocumentReference documentReference, final StringBuilder commentBuilder) {
+        final CodeableConcept codeableConcept = documentReference.getType();
+        final Optional<CodeableConcept> type = Optional.of(codeableConcept);
+
+        type.filter(CodeableConcept::hasText)
+            .map(CodeableConcept::getText)
+            .ifPresentOrElse(text -> commentBuilder.append("Type: ").append(text).append(StringUtils.SPACE),
```

Comment:
there is util method for getting text or display CodeableConceptMappingUtils, it would be worth to use it here also.

Reply:
here and the same with settings.

Reply:
I wasn't aware of this so thank you for the suggestion 😄